test/functional/diff/screen_spec.lua

```diff
@@ -0,0 +1,785 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+
+local feed = helpers.feed
+local clear = helpers.clear
+local write_file = helpers.write_file
+
+describe('Diff mode screen', function()
+  local fname = 'Xtest-functional-diff-screen-1'
+  local fname_2 = fname .. '.2'
+  local screen
+
+  local reread = function()
+    feed(':e<cr><c-w>w:e<cr><c-w>w')
+  end
+
+  setup(function()
+    clear()
+    os.remove(fname)
+    os.remove(fname_2)
+  end)
+
+  teardown(function()
+    os.remove(fname)
+    os.remove(fname_2)
+  end)
```

Comment:
Note that teardown() runs only once. If you want to do this after _each test_, change `teardown ` to `after_each`.

Reply:
I realize that. The intent was to reuse files in all test cases within one `describe`.